### DISCUSSION TITLE
front matter results in blank line at top of output - spoiling my XML output

### DISCUSSION THREAD

**Discussion by brianloweongit** [11ty/eleventy#2564] [Q&A] [answered]
Following guides for a sitemap I've created a sitemap.njk file with front matter defining the permalink, etc., which loops though collections.all and outputs XML correctly formatted as a sitemap, except for the blank line left where the front matter was in the template.
The resulting XML is deemed invalid now, since the first line doesn't have the required XML namespace definition, etc.
The guide suggests the example code just works, and I can't find any other reference to a problem with front matter leaving blanks. Can it really be just me having this problem?
How can I suppress the blank line created?

**Comment by pdehaan** [comment, marked as answer]
Maybe try adding the Nunjucks [`trim` filter](https://mozilla.github.io/nunjucks/templating.html#trim) when outputting the `{{ content | safe | trim }}` in your layout?

**Reply by brianloweongit** [reply]
Thank you so much. I was beginning to get frustrated trying to come up with search terms that would get me an answer, and convincing myself that something must be so obvious to everyone except me that it didn't even need to be mentioned.
Thanks again.

**Reply by pdehaan** [reply]
Which XML validator were you using that was saying the XML was invalid due to whitespace above the XML namespace definition?

**Reply by brianloweongit** [reply]
Just requesting the file using Chrome (105) to check it had been rendered. Chrome shows a validation error then whatever content it found before the error. In this case none, because the error is on the first line.
